docs/backup_schedules.md

```diff
@@ -0,0 +1,69 @@
+# Backup Schedules
+
+## Additional Documentation
+* Upstream Documentation: https://velero.io/docs/v1.3.0/api-types/schedule/
+
+## Example Schedule CR
+```
+apiVersion: velero.io/v1
+kind: Schedule
+metadata:
+  name: schedule-backup-10pm
+  namespace: openshift-adp
+spec:
+  schedule: "0 22 * * ?"  
+  template:
+    includedNamespaces:
+      - test
+    ttl: 720h  # Retain backups for 30 days
+    snapshotMoveData: true
+    snapshotVolumes: true
+    storageLocation: velero-sample-1
+    volumeSnapshotLocations:
+      - velero-sample-1
+```
+
+## Example k8s cronjob to delete scheduled backups (retain latest 10 backups)
+```
+apiVersion: batch/v1
```

Comment:
How about very different approach:

1. (Optional) Create config map :
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: velero-backup-jobs
  namespace: openshift-adp
data:
  BACKUP_JOBS: |
    schedule-backup-10pm-ed-no-pvc-nisdok
    schedule-backup-10pm-ed-no-pvc-nisdok-ext
    schedule-backup-10pm-ed-no-pvc-qvistorp
```

2. The jobTemplate will be as follows (two options one using ENV and another config map from above- both are same tbh):


```yaml
apiVersion: batch/v1
kind: CronJob
metadata:
  name: oadp-backup-cleanup
  namespace: openshift-adp
spec:
  schedule: "0 23 * * *"
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: cleanup
            image: oadp-backup-cleanup-image:latest
            imagePullPolicy: Always
            env:
            - name: BACKUP_JOBS
              value: "schedule-backup-10pm-ed-no-pvc-nisdok schedule-backup-10pm-ed-no-pvc-nisdok-ext schedule-backup-10pm-ed-no-pvc-qvistorp"
          restartPolicy: OnFailure
```


```yaml
apiVersion: batch/v1
kind: CronJob
metadata:
  name: oadp-backup-cleanup
  namespace: openshift-adp
spec:
  schedule: "0 23 * * *"
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: cleanup
            image: oadp-backup-cleanup-image:latest
            imagePullPolicy: Always
            env:
            - name: BACKUP_JOBS # Above is same, just here it's controller by configmap
              valueFrom:
                configMapKeyRef:
                  name: velero-backup-jobs
                  key: BACKUP_JOBS
```

We create new cleanup image:
```dockerfile
FROM velero/velero:latest
COPY backup-cleanup.sh /scripts/backup-cleanup.sh
RUN chmod +x /scripts/backup-cleanup.sh
ENTRYPOINT ["/scripts/backup-cleanup.sh"]
```
backup-cleanup.sh
```shell
#!/bin/sh

# Ensure BACKUP_JOBS is not empty
if [ -z "$BACKUP_JOBS" ]; then
  echo "Error: BACKUP_JOBS environment variable is empty."
  exit 1
fi

# Read the BACKUP_JOBS list from the environment variable and convert it to a space-separated string
BACKUP_JOBS=$(echo "$BACKUP_JOBS" | tr '\n' ' ')

# Iterate over each job in the BACKUP_JOBS list
for JOB in $BACKUP_JOBS; do
  echo "Processing backups for job: $JOB"

  # Get backups for the job, ordered by creation date
  BACKUPS=$(velero backup get --namespace=openshift-adp --selector=velero.io/job-name="$JOB" --sort-by=.metadata.creationTimestamp -o name | cut -d'/' -f2)

  # Count the total number of backups
  BACKUP_COUNT=$(echo "$BACKUPS" | wc -l)

  # Calculate the number of backups to delete (keeping the last 10)
  DELETE_COUNT=$((BACKUP_COUNT - 10))

  # If there are backups to delete, proceed
  if [ "$DELETE_COUNT" -gt 0 ]; then
    # Get the backups to delete (oldest ones first)
    BACKUPS_TO_DELETE=$(echo "$BACKUPS" | head -n "$DELETE_COUNT")

    # Iterate over each backup to delete
    for BACKUP in $BACKUPS_TO_DELETE; do
      echo "Creating DeleteBackupRequest for backup: $BACKUP"

      # Generate a unique name for the BackupDeleteRequest
      DELETE_REQUEST_NAME="delete-$BACKUP-$(date +%s)"

      # Create the DeleteBackupRequest using kubectl (ensure the correct namespace and apiVersion)
      kubectl apply -f - <<EOF
apiVersion: velero.io/v1
kind: BackupDeleteRequest
metadata:
  name: $DELETE_REQUEST_NAME
  namespace: openshift-adp
spec:
  backupName: $BACKUP
EOF

      echo "DeleteBackupRequest created for backup: $BACKUP"
    done
  else
    echo "No backups to delete for job: $JOB"
  fi
done
```

Reply:
I like the idea.  I think both are valid and we could doc both imho.